### PR TITLE
chore(rust): report the TableNotFound error while dropping non-exist table

### PR DESCRIPTION
**Pull request by BubbleCal**
this will work after upgrading lance with https://github.com/lancedb/lance/pull/1995 merged
see #884 for details

### CODE REVIEW

**Comment by BubbleCal** [comment]
I will enable the tests after upgrading lance, for now it's blocked by some changes

File: rust/lancedb/src/connection.rs
Comment (westonpace) [inline rust/lancedb/src/connection.rs:711]: Won't this `unwrap` cause a panic since the table doesn't exist?  Should this line be removed?

**Review by westonpace — APPROVED** [review]
We should wait for the lance fixes to merge and the tests to pass before merging but the fix looks good.

File: rust/lancedb/src/connection.rs
Comment (BubbleCal) [reply rust/lancedb/src/connection.rs:711]: you are right, will fix this in the enabling PR

**Pull request merged**
